src/azul/plugins/repository/tdr_anvil/__init__.py

```diff
@@ -727,99 +733,12 @@ def convert_column(value):
-        return self.common_indexed_columns | entity_columns
-
-    common_indexed_columns = {
-        'datarepo_row_id',
-        'source_datarepo_row_ids'
-    }
-
-    # This could be consolidated with similar info from the metadata plugin?
-    indexed_columns_by_entity_type = {
-        'biosample': {
-            'biosample_id',
-            'anatomical_site',
-            'apriori_cell_type',
-            'biosample_type',
-            'disease',
-            'donor_age_at_collection_unit',
-            'donor_age_at_collection_lower_bound',
-            'donor_age_at_collection_upper_bound',
-        },
-        'dataset': {
-            'dataset_id',
-            'consent_group',
-            'data_use_permission',
-            'owner',
-            'principal_investigator',
-            'registered_identifier',
-            'title',
-            'data_modality'
-        },
-        'diagnosis': {
-            'diagnosis_id',
-            'disease',
-            'diagnosis_age_unit',
-            'diagnosis_age_lower_bound',
-            'diagnosis_age_upper_bound',
-            'onset_age_unit',
-            'onset_age_lower_bound',
-            'onset_age_upper_bound',
-            'phenotype',
-            'phenopacket'
-        },
-        'donor': {
-            'donor_id',
-            'organism_type',
-            'phenotypic_sex',
-            'reported_ethnicity',
-            'genetic_ancestry',
-            # Not stored in index; only retrieved to verify redundancy with
-            # biosample.part_of_dataset_id
-            'part_of_dataset_id'
-        },
-        'file': {
-            'file_id',
-            'data_modality',
-            'file_format',
-            'file_size',
-            'file_md5sum',
-            'reference_assembly',
-            'file_name',
-            'file_ref',
-            'is_supplementary',
-        },
-        'activity': {
-            'activity_id',
-            'activity_type',
-        },
-        'alignmentactivity': {
-            'alignmentactivity_id',
-            'activity_type',
-            'data_modality',
-            'reference_assembly',
-            # Not in schema
-            'date_created',
-        },
-        'assayactivity': {
-            'assayactivity_id',
-            'activity_type',
-            'assay_type',
-            'data_modality',
-            # Not in schema
-            'date_created',
-        },
-        'sequencingactivity': {
-            'sequencingactivity_id',
-            'activity_type',
-            'assay_type',
-            'data_modality',
-        },
-        'variantcallingactivity': {
-            'variatncallingactivity_id',
-            'activity_type',
-            'reference_assembly',
-            'data_modality'
-        }
-    }
+        table = one(
+            table for table in anvil_schema['tables']
+            if table['name'].removeprefix('anvil_') == entity_type
```

Comment:
Did you consider the possibility of there being a table called `anvil_foo` and `foo`? I believe this is actually a rather likely scenario.

Reply:
I had not.